File: Non Linear Data Structures/Trees/Javascript/Tree Traversala/DFS/preorder.js


class Node {
    constructor(data) {
        this.data = data;
        this.left = null;
        this.right = null;
    }
}

function preOrderTraversal(root) {
    if(root == null) return;

    console.log(root.data);
    preOrderTraversal(root.left);
    preOrderTraversal(root.right);
}

function preOrderTraversal(root){
	if(root == null) return;
	let st = [];
	st.push(root);
	
	while(!st.isEmpty()){
		let current = st.peek();
		st.pop();
		console.log(current.data);
		
		if(current.right != NULL) st.push(current.right);
		if(current.left != NULL) st.push(current.left)
	}
}

root = new Node(10);
root.left = new Node(11);
root.right = new Node(22);

preOrderTraversal(root);
// console.log(root.data)
// console.log(root.left.data)
// console.log(root.right.data)

File: Non Linear Data Structures/Trees/Binary Search Trees/BST Traversal/preorder.cpp

#include<iostream>
using namespace std;
class Node{
    public:
    int data;
    Node* left;
    Node* right;
    Node(int data){
        this->data = data;
        this->left = NULL;
        this->right = NULL;
    }
};

void preOrderTraversal(Node* root) {
    if(root == NULL) return;

    cout<<root->data<<"\t";
    preOrderTraversal(root->left);
    preOrderTraversal(root->right);
}

int main() {

    Node* root = new Node(50);   
    root->left = new Node(40);
    root->right = new Node(60);

    root->left->left = new Node(30);
    root->left->right = new Node(42);
    root->left->left->left = new Node(10);
    root->left->left->right = new Node(32);

    root->right->left = new Node(55);
    root->right->right = new Node(70);


    preOrderTraversal(root);

    return 0;
}
